espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/drivers/ComposeDriver.kt

```diff
@@ -0,0 +1,64 @@
+package io.appium.espressoserver.lib.drivers
+
+import androidx.compose.ui.semantics.SemanticsProperties
+import androidx.compose.ui.test.performClick
+import io.appium.espressoserver.EspressoServerRunnerTest
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidElementStateException
+import io.appium.espressoserver.lib.handlers.exceptions.NoSuchElementException
+import io.appium.espressoserver.lib.helpers.ComposeRequestBuilder
+import io.appium.espressoserver.lib.helpers.ComposeViewCache
+import io.appium.espressoserver.lib.model.AppiumParams
+import io.appium.espressoserver.lib.model.BaseElement
+import io.appium.espressoserver.lib.model.ComposeElement
+import io.appium.espressoserver.lib.model.Locator
+
+class ComposeDriver : AppDriver {
+    override fun findElement(params: Locator): BaseElement {
+        try {
+            EspressoServerRunnerTest.composeTestRule.onNode(ComposeRequestBuilder.locatorToMatcher(params))
+                .assertExists()
+        } catch (assertionError: AssertionError) {
+            throw NoSuchElementException(
+                String.format(
+                    "Could not find element with strategy %s and selector %s",
+                    params.using, params.value
+                ), assertionError
+            )
+        }
+        return ComposeElement(params)
+    }
+
+    override fun click(params: AppiumParams): Nothing? {
+        try {
+            val locator = ComposeViewCache.get(params.elementId)!!
+            EspressoServerRunnerTest.composeTestRule.onNode(ComposeRequestBuilder.locatorToMatcher(locator))
+                .performClick()
+        } catch (e: Throwable) {
```

Comment:
I would rather explicitly check the element presence in cache. Catching throwables is usually not the best approach

Reply:
same below

Reply:
changed